### PR TITLE
v1.15 Backports 2024-03-26

### PR DESCRIPTION
**Pull request by sayboras**
* [x] #31488 (@bimmlerd)
 * [x] #31345 (@pchaigno)
    * :warning: minor conflict 
 * [x] #31572 (@mhofstetter)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 31488 31345 31572
```

### CODE REVIEW

**Comment by sayboras** [comment]
/test-backport-1.15

**Review by mhofstetter — APPROVED** [review]
My commit looks good - thanks Tam.

**Review by bimmlerd — APPROVED** [review]
my PR looks good

**Comment by squeed** [comment]
Looks like there's a conflict, @sayboras

File: pkg/datapath/linux/node.go [hunk 1]
```diff
@@ -910,38 +910,8 @@ func (n *linuxNodeHandler) deleteNeighbor(oldNode *nodeTypes.Node) {
...
-
-func getV4LinkLocalIP() (net.IP, error) {
-	return getLinkLocalIP(netlink.FAMILY_V4)
-}
-
-func getV6LinkLocalIP() (net.IP, error) {
-	return getLinkLocalIP(netlink.FAMILY_V6)
+func (n *linuxNodeHandler) DeleteMiscNeighbor(oldNode *nodeTypes.Node) {
+	n.deleteNeighbor(oldNode)
 }
```
Comment (pchaigno) [inline pkg/datapath/linux/node.go]: This function pops up in `datapath: Move IPsec logic to get local IPs`, I'm guessing it's a mistake in the conflict resolution?

**Review by pchaigno — CHANGES_REQUESTED** [review]

File: pkg/datapath/linux/node.go (hunk 1 above)
Comment (sayboras) [reply pkg/datapath/linux/node.go]: ah my bad :nod:

**Comment by sayboras** [comment]
/test-backport-1.15

**Review by pchaigno — APPROVED** [review]
Thanks Tam! Looks good for my PR.

**Comment by sayboras** [comment]
/test-backport-1.15

**Comment by sayboras** [comment]
Required reviews are in, CI is green. Marking this ready to merge.

**Comment by bimmlerd** [comment]
@sayboras did you mean to add `ready-to-merge` or `don't-merge/preview`? :grin:

**Comment by sayboras** [comment]
> did you mean to add ready-to-merge or don't-merge/preview? 

ah don't merge label was added intentionally, I need to check if there is any pending work from the release perspective.

**Review by squeed — APPROVED** [review]

**Comment by squeed** [comment]
@sayboras there's a merge conflict :-(

**Comment by sayboras** [comment]
> there's a merge conflict :-(

Thanks, let me rebase again it :+1:

**Comment by sayboras** [comment]
/test-backport-1.15

**Pull request merged**